test/util/util.go

```diff
@@ -350,23 +349,30 @@ func ExpectWorkloadsToBePending(ctx context.Context, k8sClient client.Client, wl
 				continue
 			}
 			if cond.Status == metav1.ConditionFalse && cond.Reason == "Pending" {
-				pending++
+				pending.Insert(client.ObjectKeyFromObject(wl))
 			}
 		}
-		g.Expect(pending).Should(gomega.Equal(len(wls)), "Not enough workloads are pending")
+		g.Expect(pending).Should(gomega.Equal(wlKeys), "Unexpected workloads are pending")
 	}, Timeout, Interval).Should(gomega.Succeed())
 }
 
 func ExpectWorkloadsToBeAdmitted(ctx context.Context, k8sClient client.Client, wls ...*kueue.Workload) {
-	expectWorkloadsToBeAdmittedCountWithOffset(ctx, 2, k8sClient, len(wls), wls...)
+	wlKeys := workloadKeys(wls...)
```

Comment:
This is mostly temporarily duplicated ExpectWorkloadsToBeAdmittedCount. I think this function ExpectWorkloadsToBeAdmittedCount I don't like because:
1. it makes the tests intrinsically non-deterministic, I think we should avoid such tests as much as possible
2. it is currently buggy because it actually checks the count is "equal", it just makes the check very often so it usually hits the right moment 

I will open an issue to try to eliminate this function

Reply:
Opened: https://github.com/kubernetes-sigs/kueue/issues/8173